apps/src/music/views/PatternAiPanel.tsx

```diff
@@ -0,0 +1,380 @@
+import classNames from 'classnames';
+import React, {
+  ChangeEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+const aiBotImages = [
+  require(`@cdo/static/music/ai/ai-bot-0.png`),
+  require(`@cdo/static/music/ai/ai-bot-1.png`),
+  require(`@cdo/static/music/ai/ai-bot-2.png`),
+  require(`@cdo/static/music/ai/ai-bot-3.png`),
+];
+
+const aiBotImageThinking = require(`@cdo/static/music/ai/ai-bot-thinking.png`);
+
+const arrowImage = require(`@cdo/static/music/music-callout-arrow.png`);
+
+import {generatePattern} from '../ai/patternAi';
+import {PatternEventValue} from '../player/interfaces/PatternEvent';
+import MusicLibrary, {SoundData} from '../player/MusicLibrary';
+import MusicPlayer from '../player/MusicPlayer';
+
+import LoadingOverlay from './LoadingOverlay';
+import PreviewControls from './PreviewControls';
+
+import styles from './patternAiPanel.module.scss';
+
+// Generate an array containing tick numbers from 1..32.
+const arrayOfTicks = Array.from({length: 32}, (_, i) => i + 1);
+
+interface PatternAiPanelProps {
+  library: MusicLibrary;
+  initValue: PatternEventValue;
+  onChange: (value: PatternEventValue) => void;
+  previewSound: MusicPlayer['previewSound'];
+  previewPattern: MusicPlayer['previewPattern'];
+  cancelPreviews: MusicPlayer['cancelPreviews'];
+  setupSampler: MusicPlayer['setupSampler'];
+  isInstrumentLoading: MusicPlayer['isInstrumentLoading'];
+  isInstrumentLoaded: MusicPlayer['isInstrumentLoaded'];
+  registerInstrumentLoadCallback: (callback: (kit: string) => void) => void;
+}
+
+type UserCompletedTaskType = 'none' | 'generated' | 'drawnDrums';
+type GenerateStateType = 'none' | 'generating';
+
+/*
+ * Renders a UI for designing a pattern, with AI generation. This is currently
+ * used within a custom Blockly Field {@link FieldPatternAi}
+ */
+const PatternAiPanel: React.FunctionComponent<PatternAiPanelProps> = ({
```

Comment:
Discussed offline and I'm sure we'll continue to discuss, but I'm curious to see if we want to maintain two separate AI and non-AI drum panels going forward, or roll the AI features into the existing drum panel.